### ISSUE TITLE
kprove: IndexOutOfBoundsException with result from Haskell backend

### ISSUE THREAD

**Issue by ttuegel** [runtimeverification/k#551]
`kprove` throws an `IndexOutOfBoundsException` with some failing results from the Haskell backend.

Output with `--debug:`
```
java.lang.IndexOutOfBoundsException: 1
        at scala.collection.mutable.ResizableArray.apply(ResizableArray.scala:43)
        at scala.collection.mutable.ResizableArray.apply$(ResizableArray.scala:42)
        at scala.collection.mutable.ArrayBuffer.apply(ArrayBuffer.scala:48)
        at org.kframework.unparser.KOREToTreeNodes$.$anonfun$wellTyped$1(KOREToTreeNodes.scala:30)
        at org.kframework.unparser.KOREToTreeNodes$.$anonfun$wellTyped$1$adapted(KOREToTreeNodes.scala:28)
        at scala.collection.Iterator.foreach(Iterator.scala:929)
        at scala.collection.Iterator.foreach$(Iterator.scala:929)
        at scala.collection.AbstractIterator.foreach(Iterator.scala:1417)
        at scala.collection.IterableLike.foreach(IterableLike.scala:71)
        at scala.collection.IterableLike.foreach$(IterableLike.scala:70)
        at scala.collection.AbstractIterable.foreach(Iterable.scala:54)
        at org.kframework.unparser.KOREToTreeNodes$.wellTyped(KOREToTreeNodes.scala:28)
        at org.kframework.unparser.KOREToTreeNodes$.$anonfun$apply$3(KOREToTreeNodes.scala:48)
        at org.kframework.unparser.KOREToTreeNodes$.$anonfun$apply$3$adapted(KOREToTreeNodes.scala:48)
        at scala.collection.immutable.HashSet$HashSet1.filter0(HashSet.scala:317)
        at scala.collection.immutable.HashSet$HashTrieSet.filter0(HashSet.scala:933)
        at scala.collection.immutable.HashSet$HashTrieSet.filter0(HashSet.scala:933)
        at scala.collection.immutable.HashSet.filter(HashSet.scala:168)
        at scala.collection.immutable.HashSet.filter(HashSet.scala:34)
        at org.kframework.unparser.KOREToTreeNodes$.apply(KOREToTreeNodes.scala:48)
        at org.kframework.unparser.KOREToTreeNodes$.$anonfun$apply$1(KOREToTreeNodes.scala:45)
        at scala.collection.TraversableLike.$anonfun$map$1(TraversableLike.scala:234)
        at scala.collection.Iterator.foreach(Iterator.scala:929)
        at scala.collection.Iterator.foreach$(Iterator.scala:929)
        at scala.collection.AbstractIterator.foreach(Iterator.scala:1417)
        at scala.collection.IterableLike.foreach(IterableLike.scala:71)
        at scala.collection.IterableLike.foreach$(IterableLike.scala:70)
        at scala.collection.AbstractIterable.foreach(Iterable.scala:54)
        at scala.collection.TraversableLike.map(TraversableLike.scala:234)
        at scala.collection.TraversableLike.map$(TraversableLike.scala:227)
        at scala.collection.AbstractTraversable.map(Traversable.scala:104)
        at org.kframework.unparser.KOREToTreeNodes$.apply(KOREToTreeNodes.scala:45)
        at org.kframework.unparser.KOREToTreeNodes.apply(KOREToTreeNodes.scala)
        at org.kframework.unparser.KPrint.unparseInternal(KPrint.java:185)
        at org.kframework.unparser.KPrint.access$000(KPrint.java:47)
        at org.kframework.unparser.KPrint$1.apply(KPrint.java:211)
        at org.kframework.unparser.KPrint$1.apply(KPrint.java:200)
        at org.kframework.kore.KTransformer.apply(transformers.scala:11)
        at org.kframework.kore.KTransformer.apply$(transformers.scala:10)
        at org.kframework.kore.AbstractKTransformer.apply(transformers.scala:123)
        at org.kframework.unparser.KPrint.sortCollections(KPrint.java:220)
        at org.kframework.unparser.KPrint.abstractTerm(KPrint.java:189)
        at org.kframework.unparser.KPrint.prettyPrint(KPrint.java:137)
        at org.kframework.unparser.KPrint.prettyPrint(KPrint.java:125)
        at org.kframework.unparser.KPrint.prettyPrint(KPrint.java:107)
        at org.kframework.unparser.KPrint.prettyPrint(KPrint.java:99)
        at org.kframework.kprove.KProve.run(KProve.java:70)
        at org.kframework.kprove.KProveFrontEnd.run(KProveFrontEnd.java:92)
        at org.kframework.main.FrontEnd.main(FrontEnd.java:59)
        at org.kframework.main.Main.runApplication(Main.java:118)
        at org.kframework.main.Main.runApplication(Main.java:107)
        at org.kframework.main.Main.main(Main.java:53)
[Error] Internal: Uncaught exception thrown of type IndexOutOfBoundsException
(1)
```

`result.kore`:
```
\and{SortGeneratedTopCell{}}(
    Lbl'-LT-'generatedTop'-GT-'{}(
        Lbl'-LT-'kevm'-GT-'{}(
            Lbl'-LT-'k'-GT-'{}(
                kseq{}(
                    inj{SortInt{}, SortKItem{}}(
                        Lbl'Hash'blockhash'LParUndsCommUndsCommUndsCommUndsRParUnds'EVM'UndsUnds'List'Unds'Int'Unds'Int'Unds'Int{}(
                            VarBLOCK'Unds'HASH:SortList{},
                            VarR'Unds'W0:SortInt{},
                            Lbl'Unds'-Int'UndsUnds'INT-COMMON'UndsUnds'Int'Unds'Int{}(
                                VarNUMBER'Unds'CELL:SortInt{},
                                \dv{SortInt{}}("1")
                            ),
                            \dv{SortInt{}}("0")
                        )
                    ),
                    kseq{}(
                        inj{SortInternalOp{}, SortKItem{}}(
                            Lbl'Hash'push'Unds'EVM'Unds'{}()
                        ),
                        kseq{}(
                            inj{SortInternalOp{}, SortKItem{}}(
                                Lbl'Hash'pc'LSqBUndsRSqBUnds'EVM'UndsUnds'OpCode{}(
                                    inj{SortUnStackOp{}, SortOpCode{}}(
                                        LblBLOCKHASH'Unds'EVM'Unds'{}()
                                    )
                                )
                            ),
                            VarDotVar2:SortK{}
                        )
                    )
                )
            ),
            Var'Unds'33:SortExitCodeCell{},
            Var'Unds'34:SortModeCell{},
            Lbl'-LT-'schedule'-GT-'{}(VarSCHEDULE:SortSchedule{}),
            Lbl'-LT-'ethereum'-GT-'{}(
                Lbl'-LT-'evm'-GT-'{}(
                    Var'Unds'10:SortOutputCell{},
                    Var'Unds'11:SortStatusCodeCell{},
                    Var'Unds'12:SortCallStackCell{},
                    Var'Unds'13:SortInterimStatesCell{},
                    Var'Unds'14:SortTouchedAccountsCell{},
                    Lbl'-LT-'callState'-GT-'{}(
                        Var'Unds'0:SortProgramCell{},
                        Var'Unds'1:SortProgramBytesCell{},
                        Var'Unds'2:SortIdCell{},
                        Var'Unds'3:SortCallerCell{},
                        Var'Unds'4:SortCallDataCell{},
                        Var'Unds'5:SortCallValueCell{},
                        Lbl'-LT-'wordStack'-GT-'{}(
                            VarR'Unds'WS:SortWordStack{}
                        ),
                        Var'Unds'6:SortLocalMemCell{},
                        Lbl'-LT-'pc'-GT-'{}(VarPCOUNT:SortInt{}),
                        Lbl'-LT-'gas'-GT-'{}(
                            Lbl'Unds'-Int'UndsUnds'INT-COMMON'UndsUnds'Int'Unds'Int{}(
                                VarG:SortInt{},
                                Lbl'Unds-LT-Unds-GT-Unds'EVM'UndsUnds'ScheduleConst'Unds'Schedule{}(
                                    LblGblockhash'Unds'EVM'Unds'{}(),
                                    VarSCHEDULE:SortSchedule{}
                                )
                            )
                        ),
                        Var'Unds'7:SortMemoryUsedCell{},
                        Var'Unds'8:SortCallGasCell{},
                        Lbl'-LT-'static'-GT-'{}(VarSTATIC:SortBool{}),
                        Var'Unds'9:SortCallDepthCell{}
                    ),
                    Var'Unds'15:SortSubstateCell{},
                    Var'Unds'16:SortGasPriceCell{},
                    Var'Unds'17:SortOriginCell{},
                    Var'Unds'18:SortPreviousHashCell{},
                    Var'Unds'19:SortOmmersHashCell{},
                    Var'Unds'20:SortCoinbaseCell{},
                    Var'Unds'21:SortStateRootCell{},
                    Var'Unds'22:SortTransactionsRootCell{},
                    Var'Unds'23:SortReceiptsRootCell{},
                    Var'Unds'24:SortLogsBloomCell{},
                    Var'Unds'25:SortDifficultyCell{},
                    Lbl'-LT-'number'-GT-'{}(VarNUMBER'Unds'CELL:SortInt{}),
                    Var'Unds'26:SortGasLimitCell{},
                    Var'Unds'27:SortGasUsedCell{},
                    Var'Unds'28:SortTimestampCell{},
                    Var'Unds'29:SortExtraDataCell{},
                    Var'Unds'30:SortMixHashCell{},
                    Var'Unds'31:SortBlockNonceCell{},
                    Var'Unds'32:SortOmmerBlockHeadersCell{},
                    Lbl'-LT-'blockhash'-GT-'{}(VarBLOCK'Unds'HASH:SortList{})
                ),
                VarDotVar3:SortNetworkCell{}
            )
        ),
        VarDotVar0:SortGeneratedCounterCell{}
    ),
    \and{SortGeneratedTopCell{}}(
        \and{SortGeneratedTopCell{}}(
            \and{SortGeneratedTopCell{}}(
                \and{SortGeneratedTopCell{}}(
                    \equals{SortBool{}, SortGeneratedTopCell{}}(
                        Lbl'Unds'andBool'Unds'{}(
                            Lbl'Unds-GT-Eqls'Int'UndsUnds'INT-COMMON'UndsUnds'Int'Unds'Int{}(
                                VarG:SortInt{},
                                Lbl'Unds-LT-Unds-GT-Unds'EVM'UndsUnds'ScheduleConst'Unds'Schedule{}(
                                    LblGblockhash'Unds'EVM'Unds'{}(),
                                    VarSCHEDULE:SortSchedule{}
                                )
                            ),
                            LblnotBool'Unds'{}(
                                Lbl'Unds'orBool'UndsUnds'BOOL'UndsUnds'Bool'Unds'Bool{}(
                                    Lbl'Unds-LT-'Int'UndsUnds'INT-COMMON'UndsUnds'Int'Unds'Int{}(
                                        Lbl'Hash'sizeWordStack'LParUndsCommUndsRParUnds'EVM-DATA'UndsUnds'WordStack'Unds'Int{}(
                                            VarR'Unds'WS:SortWordStack{},
                                            \dv{SortInt{}}("1")
                                        ),
                                        \dv{SortInt{}}("1")
                                    ),
                                    Lbl'Unds-GT-'Int'UndsUnds'INT-COMMON'UndsUnds'Int'Unds'Int{}(
                                        Lbl'UndsPlus'Int'Unds'{}(
                                            Lbl'Hash'sizeWordStack'LParUndsCommUndsRParUnds'EVM-DATA'UndsUnds'WordStack'Unds'Int{}(
                                                VarR'Unds'WS:SortWordStack{},
                                                \dv{SortInt{}}("1")
                                            ),
                                            \dv{SortInt{}}("0")
                                        ),
                                        \dv{SortInt{}}("1024")
                                    )
                                )
                            )
                        ),
                        \dv{SortBool{}}("true")
                    ),
                    \equals{SortBool{}, SortGeneratedTopCell{}}(
                        Lbl'Unds'andBool'Unds'{}(
                            LblnotBool'Unds'{}(
                                Lbl'Unds'orBool'UndsUnds'BOOL'UndsUnds'Bool'Unds'Bool{}(
                                    Lbl'Unds-LT-'Int'UndsUnds'INT-COMMON'UndsUnds'Int'Unds'Int{}(
                                        Lbl'Hash'sizeWordStack'LParUndsCommUndsRParUnds'EVM-DATA'UndsUnds'WordStack'Unds'Int{}(
                                            VarR'Unds'WS:SortWordStack{},
                                            \dv{SortInt{}}("1")
                                        ),
                                        \dv{SortInt{}}("1")
                                    ),
                                    Lbl'Unds-GT-'Int'UndsUnds'INT-COMMON'UndsUnds'Int'Unds'Int{}(
                                        Lbl'UndsPlus'Int'Unds'{}(
                                            Lbl'Hash'sizeWordStack'LParUndsCommUndsRParUnds'EVM-DATA'UndsUnds'WordStack'Unds'Int{}(
                                                VarR'Unds'WS:SortWordStack{},
                                                \dv{SortInt{}}("1")
                                            ),
                                            \dv{SortInt{}}("0")
                                        ),
                                        \dv{SortInt{}}("1024")
                                    )
                                )
                            ),
                            LblnotBool'Unds'{}(
                                Lbl'Unds'andBool'Unds'{}(
                                    VarSTATIC:SortBool{},
                                    \dv{SortBool{}}("false")
                                )
                            )
                        ),
                        \dv{SortBool{}}("true")
                    )
                ),
                \not{SortGeneratedTopCell{}}(
                    \equals{SortBool{}, SortGeneratedTopCell{}}(
                        Lbl'Unds-LT-'Int'UndsUnds'INT-COMMON'UndsUnds'Int'Unds'Int{}(
                            VarG:SortInt{},
                            Lbl'Unds-LT-Unds-GT-Unds'EVM'UndsUnds'ScheduleConst'Unds'Schedule{}(
                                LblGblockhash'Unds'EVM'Unds'{}(),
                                VarSCHEDULE:SortSchedule{}
                            )
                        ),
                        \dv{SortBool{}}("true")
                    )
                )
            ),
            \equals{SortBool{}, SortGeneratedTopCell{}}(
                Lbl'Unds-GT-Eqls'Int'UndsUnds'INT-COMMON'UndsUnds'Int'Unds'Int{}(
                    VarG:SortInt{},
                    Lbl'Unds-LT-Unds-GT-Unds'EVM'UndsUnds'ScheduleConst'Unds'Schedule{}(
                        LblGblockhash'Unds'EVM'Unds'{}(),
                        VarSCHEDULE:SortSchedule{}
                    )
                ),
                \dv{SortBool{}}("true")
            )
        ),
        \not{SortGeneratedTopCell{}}(
            \ceil{SortInt{}, SortGeneratedTopCell{}}(
                Lbl'Hash'blockhash'LParUndsCommUndsCommUndsCommUndsRParUnds'EVM'UndsUnds'List'Unds'Int'Unds'Int'Unds'Int{}(
                    VarBLOCK'Unds'HASH:SortList{},
                    VarR'Unds'W0:SortInt{},
                    Lbl'Unds'-Int'UndsUnds'INT-COMMON'UndsUnds'Int'Unds'Int{}(
                        VarNUMBER'Unds'CELL:SortInt{},
                        \dv{SortInt{}}("1")
                    ),
                    \dv{SortInt{}}("0")
                )
            )
        )
    )
)
```

**Comment by dwightguth** [comment]
I won't be able to debug this issue without a way to reproduce it, unless you wanted to run this in intellij and tell me what the value of p in KOREToTreeNodes.wellTyped and t in KOREToTreeNodes.apply are at the time of the exception.

**Comment by ehildenb** [comment]
Should by resolved by #552 . Re-open if not.
